public/app/plugins/gen.go

```diff
@@ -91,11 +92,11 @@ func adaptToPipeline(j codejen.OneToOne[corecodegen.SchemaForGen]) codejen.OneTo
 	})
 }
 
-func toDeclForGen(j codejen.OneToOne[*corecodegen.DefForGen]) codejen.OneToOne[*pfs.PluginDecl] {
-	return codejen.AdaptOneToOne(j, func(pd *pfs.PluginDecl) *corecodegen.DefForGen {
-		kd, err := corecodegen.ForGen(pd.Lineage.Runtime(), pd.KindDecl.Some())
+func kind2pd(j codejen.OneToOne[kindsys.Kind]) codejen.OneToOne[*pfs.PluginDecl] {
+	return codejen.AdaptOneToOne(j, func(pd *pfs.PluginDecl) kindsys.Kind {
+		kd, err := kindsys.BindComposable(nil, pd.KindDecl)
 		if err != nil {
-			panic("should be unreachable")
+			return nil
```

Comment:
should this return `err`?

Reply:
let me double check, but where i landed is that it shouldn't because it's expected that these implementations are sometimes empty

Reply:
yep, confirmed